src/network/networkd-link.c

```diff
@@ -340,9 +307,11 @@ static int link_enable_ipv6(Link *link) {
         if (enabled) {
                 r = sysctl_write_ip_property_boolean(AF_INET6, link->ifname, "disable_ipv6", false);
                 if (r < 0)
-                        log_link_warning_errno(link, r, "Cannot enable IPv6: %m");
-                else
-                        log_link_info(link, "IPv6 successfully enabled");
+                        return log_link_warning_errno(link, r, "Cannot enable IPv6: %m");
+
+                log_link_info(link, "IPv6 successfully enabled");
+                /* We know what the sysctl value is, so update our internal cache. */
+                link->sysctl_ipv6_enabled = true;
```

Comment:
OK, then please drop `Link::sysctl_ipv6_enabled`. It is not used anymore.

Reply:
Thanks. Updated.